apollo-backend/src/main/java/io/logz/apollo/services/BlockerService.java

```diff
@@ -236,4 +237,26 @@ public RequestBlockerResponse checkDeploymentShouldBeBlockedBySingleRegionBlocke
         return new RequestBlockerResponse(false, SingleRegionBlocker.BLOCKER_NAME);
     }
 
+    public Optional<String> initSingleRegionBlockerConfiguration(Integer serviceId, Integer stackId) {
+        if (serviceId != null) {
+            return Optional.of(getSingleRegionBlockerConfiguration(new ArrayList<Integer>() {{
+                add(serviceId);
+            }}));
+        } else if (stackId != null) {
+            List<Integer> serviceIds = blockerInjectableCommons.getStackDao().getServicesStack(stackId).getServices();
+            if (!serviceIds.isEmpty()) {
+                return Optional.of(getSingleRegionBlockerConfiguration(serviceIds));
+            }
+        }
+
+        return Optional.empty();
+
+    }
+
+    private String getSingleRegionBlockerConfiguration(List<Integer> serviceIds) {
+        return "{\n" +
+                "  \"serviceIds\":" + serviceIds.toString() +
+                "}";
+    }
+
 }
```

Comment:
maybe better to move it to BlockerDefinitionController? Regular blockers are fully defined in BlockerDefinitionController..

Reply:
Maty says that it is better to have the logic on blockerService the Controller - and I am agree. If you want I can do a big refactor on all the Apollo Project.